homeassistant/helpers/update_coordinator.py

```diff
@@ -444,20 +455,9 @@ class TimestampDataUpdateCoordinator(DataUpdateCoordinator[_DataT]):
 
     last_update_success_time: datetime | None = None
 
-    async def _async_refresh(
-        self,
-        log_failures: bool = True,
-        raise_on_auth_failed: bool = False,
-        scheduled: bool = False,
-        raise_on_entry_error: bool = False,
-    ) -> None:
-        """Refresh data."""
-        await super()._async_refresh(
-            log_failures,
-            raise_on_auth_failed,
-            scheduled,
-            raise_on_entry_error,
-        )
+    @callback
+    def _async_refresh_finished(self) -> None:
+        """Handle when a refresh has finished."""
```

Comment:
We would set `last_update_success_time` too late and than end up in a loop because the listeners where dispatched before the value was set

Reply:
This break the loop, but I'm also wondering if we have an infinite loop in production on failure currently and it doesn't look like we have tests for that case

Reply:
failure case doesn't loop

https://github.com/home-assistant/core/pull/115541